jina/serve/networking.py

```diff
@@ -466,20 +467,21 @@ async def send_requests(
                             return response, None
 
             if request_type == DataRequest and len(requests) > 1:
-                # TODO: This section of the code has no observability
-                # TODO: Does this section not need to deal with streaming stubs?
                 if self.data_list_stub:
                     call_result = self.data_list_stub.process_data(
                         requests,
                         metadata=metadata,
                         compression=compression,
                         timeout=timeout,
                     )
+                    for response in requests:
```

Comment:
why don't u do it before the response? and why u iterate as `response` and not `request`?

Reply:
Oh I copied it from the loop in stream_stub. Didnt notice the iter var is named wrongly

Reply:
> why don't u do it before the response?

This is again copied from above from line 445